### PR TITLE
[JENKINS-61760] Use git plugin's MergeWithSCMExtension Class

### PR DESCRIPTION
**Pull request by baymac**
Unable to use git plugin 4.2.2 :(

### CODE REVIEW

**Comment by jetersen** [comment]
Could be a good time to switch to `bom` to avoid all these conflict :)
https://github.com/jenkinsci/git-plugin/blob/580f57853880192353c9e33adc01ee2f7224bac3/pom.xml#L276-L286
https://github.com/jenkinsci/bom

**Comment by baymac** [comment]
@jetersen Can you take a look at pom now? Also unable unable to upgrade JCasC version.

File: pom.xml [hunk 1]
```diff
@@ -44,7 +44,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>scm-api</artifactId>
-      <version>2.5.1</version>
+      <version>2.6.3</version>
```
Comment (jetersen) [inline pom.xml:47]: The idea is to remove the version number actually and let it come from the BOM 👍

Comment (baymac) [reply pom.xml]: Fixed in bf0fb77

**Comment by baymac** [comment]
Spot bugs doesn't allow to use same class name as super? Should I just remove the `MergeWithGitSCMExtension` class from this plugin? @jetersen

**Comment by jetersen** [comment]
There used the opportunity to clean up the pom and update to the new jcasc test-harness

**Comment by jetersen** [comment]
Feel free to squash the PR 😓

**Pull request merged**